homeassistant/components/light/mqtt/__init__.py

```diff
@@ -0,0 +1,62 @@
+"""
+Support for MQTT lights.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/light.mqtt/
+"""
+import logging
+
+from homeassistant.components import mqtt, light
+from homeassistant.components.mqtt import ATTR_DISCOVERY_HASH
+from homeassistant.components.mqtt.discovery import MQTT_DISCOVERY_NEW
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+from homeassistant.helpers.typing import HomeAssistantType, ConfigType
+
+from . import schema_basic
+from . import schema_json
+from . import schema_template
+
+_LOGGER = logging.getLogger(__name__)
+
+DEPENDENCIES = ['mqtt']
+
+PLATFORM_SCHEMA = mqtt.MQTT_RW_PLATFORM_SCHEMA.extend(
+    schema_basic.PLATFORM_SCHEMA_BASIC.schema).extend(
+        schema_json.PLATFORM_SCHEMA_JSON.schema).extend(
+            schema_template.PLATFORM_SCHEMA_TEMPLATE.schema).extend(
+                mqtt.MQTT_AVAILABILITY_SCHEMA.schema)
```

Comment:
This is not the way this should be done. If we already have a schema option, why not use it?

```python
def validate_mqtt_light(value):
    SCHEMAS = {
        'basic': schema_basic.PLATFORM_SCHEMA_BASIC,
        'json': schema_json.PLATFORM_SCHEMA_JSON,
        ...
    }
    return SCHEMAS[value[CONF_SCHEMA]](value)

PLATFORM_SCHEMA = vol.All(vol.Schema({
    vol.Optional(CONF_SCHEMA, default='basic'): vol.All(vol.Lower, vol.Any('basic', 'json', 'template'))
}, allow_extra=True), validate_mqtt_light)
```

Reply:
Thanks!
However, what should the custom validator do if it finds a violation?
Print error messsage? Throw?

Reply:
Thanks a lot!